shared/directives/drag-scroll/drag-scroll.directive.ts

```diff
@@ -0,0 +1,66 @@
+import { Directive, ElementRef, OnInit } from '@angular/core';
+import { fromEvent } from 'rxjs';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+
+@UntilDestroy()
+@Directive({
+  selector: '[appDragScroll]',
+})
+export class DragScrollDirective implements OnInit {
+  private dragScrollElement: HTMLElement;
+  private isMovable: boolean;
+  private lastEvent: MouseEvent;
+
+  constructor(
+    private elementRef: ElementRef<HTMLElement>,
+  ) {
+    this.dragScrollElement = elementRef.nativeElement;
+  }
+
+  ngOnInit(): void {
+    fromEvent(this.dragScrollElement, 'mouseover').pipe(
+      untilDestroyed(this),
+    ).subscribe(() => {
+      if (this.dragScrollElement.scrollWidth > window.innerWidth) {
+        this.dragScrollElement.style.cursor = 'grabbing';
+      }
+    });
+
+    fromEvent(this.dragScrollElement, 'mouseleave').pipe(
+      untilDestroyed(this),
+    ).subscribe(() => {
+      this.isMovable = false;
+      this.dragScrollElement.style.cursor = 'default';
```

Comment:
U can't be sure that 'default' is exactly what an element wants. It's an unsafe code anyway, but save the cursor state using `ngOnInit()`, may be `cursor: pointer` is an initial state